puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/tarballsRelease.groovy

```diff
@@ -0,0 +1,106 @@
+    def major_version = env.getProperty('major_version')
+    def ruby_ver = '2.4.0'
+
+    stages {
+        stage('Parallel') {
+
+            parallel {
+                stage('Foreman') {
+                    steps {
+                        verify_tag('foreman', version)
+                        build_tarball('foreman', version, ruby_ver)
+                    }
+                }
+                stage('Smart Proxy') {
+                    steps {
+                        verify_tag('smart-proxy', version)
+                        build_tarball('smart-proxy', version, ruby_ver)
+                    }
+                }
+                stage('Foreman SELinux') {
+                    steps {
+                        verify_tag('foreman-selinux', version)
+                        build_tarball('foreman-selinux', version, ruby_ver)
+                    }
+                }
+                stage('Foreman Installer') {
+                    steps {
+                        verify_tag('foreman-installer', version)
+                        build_tarball('foreman-installer', version, ruby_ver)
+                    }
+                }
+            }
+        }
+    }
+
+    post {
+        success {
+            build job: 'release_packages', parameters: [
+                [$class: 'StringParameterValue', name: 'version', value: version],
+                [$class: 'StringParameterValue', name: 'major_version', value: major_version]
+            ]
+        }
+    }
+}
+
+void verify_tag(project, version) {
+    deleteDir()
+    dir(project) {
+        git url: "https://github.com/theforeman/${project}", branch: 'develop'
+        sh """
+        if [ \"$(git tag -l ${version} | wc -l)\" -ne 1 ]; then
+            echo \"ERROR: cannot find tag ${version}\"
+            git tag -l
+            exit 1
+        fi
+        exit 0
+        """
+    }
+    deleteDir()
+}
+
+void build_tarball(project, version, ruby_ver) {
+    def base_dir = "/var/www/vhosts/downloads/htdocs/foreman"
+    def rake = "rake"
+
+    deleteDir()
+    dir(project) {
+        git url: 'https://github.com/theforeman/foreman', branch: version
+        configureRVM(ruby_ver)
+
+        if (project == 'foreman') {
+            sh "cat config/settings.yaml.example > config/settings.yaml"
+            sh "cat config/database.yml.example > config/database.yml"
+        }
+
+        if (project == 'foreman-installer') {
+            env.setProperty('PUPPET_VERSION', '4.10')
+        }
+
+        env.setProperty('DEBUG_RESOLVER', '1')
+
+        if (fileExists("Gemfile")) {
+            withRVM(["bundle install --without=development --retry=5"], ruby_ver)
+            rake = "bundle exec rake"
+        }
+
+        withRVM(["${rake} pkg:generate_source", ruby_ver)
+
+        sh "ssh root@theforeman.org \"mkdir -p ${base_dir}/\" || true"
+        sh "rsync -v --ignore-existing pkg/* root@theforeman.org:${base_dir}/"
+
+        cleanupRVM('', ruby_ver)
+    }
+    deleteDir()
```

Comment:
This will similarly cause concurrency issues. I think because you scope on dir(project) this deleteDir() is unneeded.

Reply:
Would it be safer to include these **inside** the `dir(project)` block?

Reply:
Are you wanting to do the double deleteDir just to be opportunistic? Given the entire workspace is cleaned up at the end of the job these seems like overkill.

Reply:
Just trying to clean up after myself. These aren't needed?

Reply:
removed